scripts/documentation/plugin-quality-indicators.ts

```diff
@@ -0,0 +1,160 @@
+import { writeFileSync } from 'fs';
+import axios from 'axios';
+
+interface Interval {
+  start: Date;
+  end: Date;
+}
+
+interface PluginRegistry {
+  name: string;
+  description: string;
+  url: string;
+}
+
+const plugins =
+  require('../../community/approved-plugins.json') as PluginRegistry[];
+
+async function main() {
+  const qualityIndicators: any = {};
+  for (let i = 0; i < plugins.length; i++) {
+    const plugin = plugins[i];
+    const npmData = await getNpmData(plugin);
+    const npmDownloads = await getNpmDownloads(plugin);
+    const githubStars = await getGithubStars(npmData.githubRepo);
+    const nxVersion = await getNxVersion(
+      npmData.githubRepo,
+      npmData.githubDirectory
+    );
+    qualityIndicators[plugin.name] = {
+      lastPublishedDate: npmData.lastPublishedDate,
+      githubDirectory: npmData.githubDirectory,
+      npmDownloads,
+      githubStars,
+      nxVersion,
+    };
+    console.log(qualityIndicators[plugin.name]);
+  }
+
+  writeFileSync(
+    '../../nx-dev/nx-dev/pages/extending-nx/quality-indicators.json',
+    JSON.stringify(qualityIndicators, null, 2)
+  );
+}
+main();
+
+// Publish date (and github directory, readme content)
+// i.e. https://registry.npmjs.org/@nxkit/playwright
+async function getNpmData(plugin: PluginRegistry) {
+  const { data } = await axios.get(`https://registry.npmjs.org/${plugin.name}`);
+  console.log(plugin.name);
+  const lastPublishedDate = data.time[data['dist-tags'].latest];
+  if (!data.repository) {
+    console.log('!!! No repository!');
+    return { lastPublishedDate, githubRepo: '' };
+  }
+  const url: String = data.repository.url;
+  const githubRepo = url
+    .slice(url.indexOf('github.com/') + 11)
+    .replace('.git', '');
+  return {
+    lastPublishedDate,
+    githubDirectory: data.repository.directory,
+    githubRepo,
+    // readmeContent: plugin.name
+  };
+}
+
+// Download count
+// i.e. https://api.npmjs.org/downloads/point/2023-06-01:2023-07-01/@nxkit/playwright
+async function getNpmDownloads(plugin: PluginRegistry) {
+  const lastMonth = getLastMonth();
+  const { data } = await axios.get(
+    `https://api.npmjs.org/downloads/point/${stringifyIntervalForUrl(
+      lastMonth
+    )}/${plugin.name}`
+  );
+  return data.downloads;
+}
+
+export function getLastMonth() {
+  const now = new Date();
+  const thisYear = now.getFullYear();
+  const thisMonth = now.getMonth();
+  return {
+    start: new Date(`${thisYear}-${thisMonth - 1}-01`),
+    end: new Date(`${thisYear}-${thisMonth}-01`),
+  };
+}
```

Comment:
So this is the last month? So on August 31st... it would be downloads from July 1 to August 1?

Can we do a rolling month? Or Last 30 days?

Reply:
The last 30 days make the most sense.